### PR TITLE
WIP -- models

### PR DESCRIPTION
**Pull request by phuchle**
I've tried to implement schemas and models based on Alex's UML diagram but I'm at the point where I'm in over my head.  I'm creating this PR to get feedback and some direction on what to do next.

For reference:
[Adopt-A-Family_Model_Diagram.pdf](https://github.com/5-gwoap/adopt-a-family/files/1490350/Adopt-A-Family_Model_Diagram.pdf)

### CODE REVIEW

**Comment by jwu910** [comment]
You've got conflicts on here. Did you rebase on to updated development to get changes form the last merge? 

Edit: Detective efrain said this was due to one of my bad rebases. Apologies.

File: server/models/donor.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const mongoose = require('mongoose');
```
Comment (jwu910) [inline server/models/donor.js:1]: Question:
When do we use es6 `Import` vs `const = require();`?

Comment (phuchle) [reply server/models/donor.js:1]: In our case, `import` in the react project and `require()` in the rest of it.

Comment (jwu910) [reply server/models/donor.js:1]: :+1: I think i understand. lol

Comment (phuchle) [reply server/models/donor.js:1]: `import` isn't implemented in node.js yet and we are able to use it in react, thanks to webpack + babel-plugins.

File: server/models/user.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+const bcrypt = require('bcrypt');
+
+const saltRounds = 10;
```
Comment (rainrivas) [inline server/models/user.js:5]: @phuchle is this a random encryptor?

File: server/models/organizer.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const organizerSchema = new Schema({
+  name: {
+    type: Schema.Types.ObjectId,
```
Comment (rainrivas) [inline server/models/organizer.js:6]: I see this goes to the "name" schema, is that what we want for organization, despite that name.js seems to be build for user name

**Review by rainrivas — APPROVED** [review]
Let me know if we should look into fixing organization portion or not. Also i'm merging this into a branch that is based off development so that we can tag team the models work.

**Pull request merged**